### PR TITLE
fix use of flex attn kwargs and add support for tiledmlp for GPT-OSS

### PR DESCRIPTION
**Pull request by winglian**
# Description

## Summary by CodeRabbit

- New Features
  - Added Mixture-of-Experts support to tiled MLP for gpt_oss models, enabling distributed MoE training/inference.
- Bug Fixes
  - Improved FlexAttention stability and compatibility across PyTorch 2.5.1 and 2.6.0 via version-aware compilation to reduce compile/runtime issues.
  - Fixed gradient handling for tiled MLP with sharded execution, ensuring correct accumulation across shards and proper support for tuple outputs.

### CODE REVIEW

File: src/axolotl/monkeypatch/attention/flex_attn.py [hunk 1]
```diff
@@ -46,19 +47,25 @@ def __init__(self, training):
...
+                self.training = training
+                if is_torch_less_or_equal("2.5.1"):
+                    self._compiled_flex_attention = torch.compile(
+                        flex_attention, dynamic=False
+                    )
                 # In PyTorch 2.6.0, there's a known issue with flex attention compilation which may
                 # cause errors. The suggested fix is to compile with "max-autotune-no-cudagraphs"
                 # see https://github.com/pytorch/pytorch/issues/146260 for training
-                self.training = training
-                LOG.info(
```
Comment (salmanmohammadi) [inline src/axolotl/monkeypatch/attention/flex_attn.py:53]: I found this helpful as sometimes compilation took a while and it was good to know training wasn't stuck
